torch/csrc/jit/codegen/cuda/python_frontend/fusion_cache.cpp

```diff
@@ -19,6 +19,7 @@ FusionCache* FusionCache::get(size_t max_fusions) {
   if (singleton_ == nullptr) {
     singleton_ = new FusionCache(max_fusions);
   }
+  singleton_->max_fusions_ = max_fusions;
```

Comment:
Should we check `fusions_.size() <= max_fuions`?

Reply:
Yes, that is a good idea.  I added the check.